src/treeview/writer.js

```diff
@@ -833,9 +856,38 @@ function unwrapAttributeElement( wrapper, toUnwrap ) {
 // (`start` and `end` positions are located inside same {@link engine.treeView.AttributeElement AttributeElement}),
 // starts on 0 offset and ends after last child node.
 //
+// @private
 // @param {engine.treeView.Range} Range
 // @returns {Boolean}
 function rangeSpansOnAllChildren( range ) {
 	return range.start.parent == range.end.parent && range.start.parent instanceof AttributeElement &&
 		range.start.offset === 0 && range.end.offset === range.start.parent.getChildCount();
+}
+
+// Checks if provided nodes are valid to insert by writer. Checks if each node is an instance of
+// {@link engine.treeView.Text Text} or {@link engine.treeView.AttributeElement AttributeElement} or
+// {@link engine.treeView.ContainerElement ContainerElement}.
+//
+// Throws {@link utils.CKEditorError CKEditorError} `treeview-writer-insert-invalid-node` when nodes to insert
+// contains instances that are not {@link engine.treeView.Text Texts},
+// {@link engine.treeView.AttributeElement AttributeElements} or
+// {@link engine.treeView.ContainerElement ContainerElements}.
+//
+// @private
+// @param Iterable.<engine.treeView.Text|engine.treeView.AttributeElement|engine.treeView.ContainerElement> nodes
+function validateNodesToInsert( nodes ) {
+	for ( let node of nodes ) {
```

Comment:
Right. So the doc is wrong. But if you change the doc then the `validateNodesToInsert( node.getChildren() )` call below will be incorrect :D.